src/cloud.c

```diff
@@ -283,6 +283,10 @@ int cloud_register_device(const char *id, const char *name)
 	}
 
 	jobj_device = parser_device_json_create(id, name);
+	if (!jobj_device) {
+		amqp_bytes_free(queue_cloud);
+		return -1;
```

Comment:
Since this code will be sent to things. We should use one of the knot code err.
As an example `KNOT_ERR_CLOUD_FAILURE`

Reply:
Ok